core/src/services/routing.js

```diff
@@ -404,6 +404,20 @@ class RoutingClass {
       }
 
       let cNode2 = currentNode;
+      let hideGlobalSearchInherited = nodeObject.hideGlobalSearch;
+      if (hideGlobalSearchInherited === undefined) {
+        while (cNode2) {
+          if (cNode2.tabNav && cNode2.hideGlobalSearch === true) {
+            hideGlobalSearchInherited = true;
+            break;
+          }
+          if (cNode2.hideGlobalSearch === false) {
+            hideGlobalSearchInherited = false;
+            break;
+          }
+          cNode2 = NavigationHelpers.getParentNode(cNode2, pathData.navigationPath);
+        }
+      }
```

Comment:
Could we extract this logic in a function of its own and add some documentation to it with JSDoC. This function is super long and lets not have it become even longer 😄  . Also, if we plan to remove this feature in the future, it would be also much easier to remove, by just deleting the function thereafter.